File: app/Models/Enneagram.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Enneagram extends Model
{
    use HasFactory;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'enneagrams';

    /**
     * Indicates if the model should be timestamped.
     * The schema does not have created_at/updated_at columns.
     *
     * @var bool
     */
    public $timestamps = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'type_number',
        'name',
        'description',
        'attributes',
    ];

    /**
     * The attributes that should be cast.
     * Automatically converts the JSONB column to and from a PHP array.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'attributes' => 'array',
    ];

    /**
     * The users that have been assigned this Enneagram type.
     * Defines the inverse of the many-to-many relationship.
     */
    public function users()
    {
        return $this->belongsToMany(User::class, 'user_enneagrams', 'enneagram_id', 'user_id');
    }
}

File: python-backend/app/models/enneagram.py
"""
Enneagram model.
"""
from sqlalchemy import Column, Integer, SmallInteger, Text
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import relationship

from app.core.database import Base


class Enneagram(Base):
    """Enneagram personality type model."""
    
    __tablename__ = "enneagrams"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    type_number = Column(SmallInteger, unique=True, nullable=False)
    name = Column(Text, nullable=False)
    description = Column(Text)
    attributes = Column(JSONB)
    
    # Relationships
    users = relationship(
        "User",
        secondary="user_enneagrams",
        back_populates="enneagrams"
    )
    
    def __repr__(self):
        return f"<Enneagram Type {self.type_number}: {self.name}>"

